MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2257,11 +2253,16 @@ public string GetAudioFilterParam(EncodingJobInfo state, EncodingOptions encodin
                     : inputChannels.Value;
             }
 
-            if (isTranscodingAudio && transcoderChannelLimit.HasValue)
+            // Set resultChannels to minimum between resultChannels, TranscodingMaxAudioChannels, transcoderChannelLimit
+            if (isTranscodingAudio)
             {
                 resultChannels = resultChannels.HasValue
-                    ? Math.Min(resultChannels.Value, transcoderChannelLimit.Value)
-                    : transcoderChannelLimit.Value;
+                    ? Math.Min(resultChannels.Value, request.TranscodingMaxAudioChannels.HasValue
+                               ? Math.Min(request.TranscodingMaxAudioChannels.Value, transcoderChannelLimit.Value)
+                               : transcoderChannelLimit.Value)
+                    : request.TranscodingMaxAudioChannels.HasValue
+                        ? Math.Min(request.TranscodingMaxAudioChannels.Value, transcoderChannelLimit.Value)
+                        : transcoderChannelLimit.Value;
```

Comment:
```cs
resultChannels = Math.Min(resultChannels.Value, transcoderChannelLimit.Value);
if (request.TranscodingMaxAudioChannels.HasValue && request.TranscodingMaxAudioChannels.Value < resultChannels)
{
    resultChannels = request.TranscodingMaxAudioChannels.Value;
}
```

Reply:
This does not take into account if resultChannels does not have a value, which the previous code did. Is this still something to worry about?

Reply:
I've refactored this. Please take another look when you are able; if you'd prefer I use if statements instead of the ternary ones, I can switch, but I thought this made it a bit more readable because of the nullable types.